include/irr/asset/ICPUShader.h

```diff
@@ -35,19 +38,16 @@ class ICPUShader : public IAsset, public IShader<ICPUBuffer>
         core::smart_refctd_ptr<IAsset> clone(uint32_t _depth = ~0u) const override
         {
             auto buf = (_depth > 0u && m_code) ? core::smart_refctd_ptr_static_cast<ICPUBuffer>(m_code->clone(_depth-1u)) : m_code;
-            auto cp = core::make_smart_refctd_ptr<ICPUShader>(std::move(buf));
-
-            cp->m_containsGLSL = m_containsGLSL;
-
-            cp->m_mutable = true;
+            auto cp = core::smart_refctd_ptr<ICPUShader>(new ICPUShader(std::move(buf), m_containsGLSL), core::dont_grab);
+            clone_common(cp.get());
```

Comment:
I think you lost `m_containsGLSL` here...

Reply:
nah, see constructor